src/NServiceBus.RavenDB/Internal/DocumentStoreManager.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        private static DocumentStoreInitContext CreateDefaultDocumentStore(ReadOnlySettings settings)
+        {
+            var p = settings.GetOrDefault<ConnectionParameters>(RavenDbSettingsExtensions.DefaultConnectionParameters);
+            if (p != null)
+            {
+                var storeByParams = new DocumentStore
+                {
+                    Url = p.Url,
+                    DefaultDatabase = p.DatabaseName ?? settings.EndpointName(),
+                    ApiKey = p.ApiKey,
+                    Credentials = p.Credentials
+                };
+
+                return new DocumentStoreInitContext(storeByParams);
+            }
+
+            var initContext = CreateStoreByConnectionStringName(settings, "NServiceBus/Persistence/RavenDB", "NServiceBus/Persistence");
+
+            if (initContext != null)
+            {
+                return initContext;
+            }
+
+            return CreateStoreByUrl(settings, "http://localhost:8080");
+        }
+
+        class DocumentStoreHolder
+        {
+            public DocumentStoreInitContext UninitializedDocumentStore { get; set; }
+        }
+
+        static DocumentStoreInitContext CreateStoreByConnectionStringName(ReadOnlySettings settings, params string[] connectionStringNames)
+        {
+            var connectionStringName = GetFirstNonEmptyConnectionString(connectionStringNames);
+            if (!string.IsNullOrWhiteSpace(connectionStringName))
+            {
+                var docStore = new DocumentStore
+                {
+                    ConnectionStringName = connectionStringName
+                };
+                if (docStore.DefaultDatabase == null)
+                {
+                    docStore.DefaultDatabase = settings.EndpointName();
+                }
+
+                return new DocumentStoreInitContext(docStore);
+            }
+            return null;
+        }
+
+        static DocumentStoreInitContext CreateStoreByUrl(ReadOnlySettings settings, string url)
+        {
+            var docStore = new DocumentStore
+            {
+                Url = url
+            };
+
+            if (docStore.DefaultDatabase == null)
+            {
+                docStore.DefaultDatabase = settings.EndpointName();
+            }
+
+            return new DocumentStoreInitContext(docStore);
+        }
+
+        static string GetFirstNonEmptyConnectionString(params string[] connectionStringNames)
+        {
+            try
+            {
+                return connectionStringNames.FirstOrDefault(cstr => ConfigurationManager.ConnectionStrings[cstr] != null);
+            }
+            catch (ConfigurationErrorsException)
+            {
+                return null;
+            }
+        }
+
+        static string StorageTypeToSettingsKey<TStorageType>()
+            where TStorageType : StorageType
+        {
+            if (typeof(TStorageType) == typeof(StorageType.Subscriptions))
```

Comment:
Good indirect catch. That's not referenced. That was a previous attempt before I realized that the pluralization was not even close to consistent and I had to map everything out the long way.